engine/dlib/src/wscript

```diff
@@ -63,54 +56,46 @@ def build(bld):
     if platform == 'win':
         platform = "win32"
 
-    platform_include_dir = 'dlib/%s' % platform
-
     source_files = bld.path.ant_glob(source_dirs)
 
+    platform_source_files = []
+    if platform_source_dirs:
+        platform_source_files = bld.path.ant_glob(platform_source_dirs)
+
+    # Trying to maintain this set of rules in a way that fits all platforms+combinations is tricky
+    # The current incarnation is add all common sources plus the sources for the particular platform
+    # We then remove some specific files for each platform
```

Comment:
Tried to simplify the private platform support, by being much more explicit, and using less "magic" code.